.travis.yml

```diff
@@ -60,6 +60,7 @@ matrix:
 
     - name: Verify Licenses
       script:
+        - time mvn install -T 2C -q -DskipTests=true -Dmaven.javadoc.skip=true -B -V -Dskip.npm
```

Comment:
(1) What does `-Dskip.npm` do?

(2) Why do we need to build before running `verify_licenses.sh`?

(3) Assuming we do need to build, would it make more sense to put the build step in `dev-utilities/build-utils/list_dependencies.sh`?  If I just manually run that script, we should expect it to work also, right?

Reply:
1. Just skips the UI portion of the build. it's already built-in. The catch is our build currently explodes if we do it, so a couple scripts needed adjustment. It just avoids doing the UI build work when we aren't actually looking to do it.
1. Check out the ticket itself.
1. In practice, I don't know that anyone actually runs the script without already having relevant builds (after all, messing with deps file is pretty much the last thing anybody does). We could split the difference and add a flag to run the build to the script and kick out a warning if it's not done or something?

Reply:
(3) 

> In practice, I don't know that anyone actually runs the script without already having relevant builds ...

If I get a Travis failure on the license verification, then switch to my local machine to figure out why, I would just run the script.  This could put me in a situation where the outputs are different and it would be confusing.  

It seems to have caused you some headaches in the JUnit 5 upgrade, so I think we should do more to avoid causing someone else the same headaches in the future.

> We could split the difference and add a flag to run the build to the script and kick out a warning if it's not done or something?

If you want to add a flag to the script, that works for me.  Just calling `mvn install ... ` from inside the script seems even simpler to me, but I'd be happy with a flag and check to avoid the problem.

Reply:
(1) 

> Just skips the UI portion of the build. it's already built-in. 

Is this functionally different than `-pl '!:metron-config,!:metron-alerts'` that is already used elsewhere in our `.travis.yml`?  

If it is functionally the same, then why not just use the existing approach?

Reply:
That's a good point, I don't think it's functionally different unless we were to restructure things in the future. Having said that, I'd actually make the opposite argument. If the point is to avoid npm, we should be using the built-in skip.npm and building all the modules, rather than making exceptions.

Calling maven inside the script it probably fine. I was thinking that running maven takes a lot longer than it actually does.

Reply:
> Having said that, I'd actually make the opposite argument. If the point is to avoid npm, we should be using the built-in skip.npm and building all the modules, rather than making exceptions.

I am just interested in consistency.  Using one approach is just the most obvious thing to do here and leaves less open questions for future maintainers.

If you can update the PR to use one approach or the other, I'd be a happy camper.